File: CarRental/Dtos/InvoiceDto.cs
using CarRental.Helpers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CarRental.Dtos
{
    public class InvoiceDto
    {
        public int Id { get; set; }
        public CustomerDto Customer { get; set; }
        public BookingDto Booking { get; set; }
        public string InvoiceNumber { get; set; }
        private string _issueDate { get; set; }

        public string IssueDate
        {
            get
            {
                return
                    Utility.ConvertToDatetime(_issueDate);

            }
            set { _issueDate = Utility.ConvertToDatetime(value); }

        }

        private string _dueDate { get; set; }

        public string DueDate
        {
            get
            {
                return
                    Utility.ConvertToDatetime(_dueDate);

            }
            set { _dueDate = Utility.ConvertToDatetime(value); }

        }
      
        public double Amount { get; set; }
        public string Description { get; set; }
        public int CustomerId { get; set; }
        public int BookingId { get; set; }
        public bool status { get; set; }
        public DateTime CreatedOn { get; set; }
        public string CreatedBy { get; set; }
    }
}

File: CarRental/ClientApp/src/app/classes/InvoiceDto.ts
import { CustomerDto } from "./CustomerDto";
import { BookingDto } from "./BookingDto";

 

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/

    
    export class InvoiceDto {
        
        // Id
        public Id: number = 0;
        // Customer
        public Customer: CustomerDto = null;
        // Booking
        public Booking: BookingDto = null;
        // InvoiceNumber
        public InvoiceNumber: string = null;
        // IssueDate
        public IssueDate: string = null;
        // DueDate
        public DueDate: string = null;
        // Amount
        public Amount: number = 0;
        // Description
        public Description: string = null;
        // CustomerId
        public CustomerId: number = 0;
        // BookingId
        public BookingId: number = 0;
        // status
        public status: boolean = false;
        // CreatedOn
        public CreatedOn: Date = new Date(0);
        // CreatedBy
        public CreatedBy: string = null;
    }

